helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java

```diff
@@ -147,10 +147,16 @@ protected static RealmAwareZkClient createZkClient(HelixManager manager) {
     if (Boolean.getBoolean(SystemPropertyKeys.MULTI_ZK_ENABLED) || zkAddress == null) {
```

Comment:
This probably should be pointed out in https://github.com/apache/helix/pull/1627. 

if zkAddress is null, we use federatedZkClient, which is not exactly the same as dedicated ZkClient.

My question is more like this: 
Why do we think that MULTI_ZK_ENABLED and zkAddress == null belongs tot the same handling logic? 

Or put it another way, this condition, "if zkAddress is null for the HelixManager passed to TaskStateModelFactory and multizk is not enabled"? what should be the expected behavior?

Reply:
Hi Kai, please see https://github.com/apache/helix/pull/1183 for details. In short, there are customer requirements for this type of usage.

Reply:
Let me rephrase it this way, under this situation, which ZooKeeper should we use?

Reply:
Under this condition, like all other conditions modified in #1183, simply means we're in multiZk mode. The ZooKeeper used is the one corresponding to the cluster (which is made into a sharding key).